BetaCameras/ifm/O3D3xx.cs

```diff
@@ -383,19 +433,26 @@ private void UpdateLoop()
         {
             while (!_cancelUpdateThreadSource.Token.IsCancellationRequested)
             {
-                byte[] hdrBuffer = new byte[16];
-                Receive(_clientSocket, hdrBuffer, 0, 16, 300000);
-                var str = Encoding.Default.GetString(hdrBuffer, 5, 9);
-                Int32.TryParse(str, out int frameSize);
-
-                lock (_backLock)
+                try
                 {
-                    if (_backBuffer.Length != frameSize)
+                    byte[] hdrBuffer = new byte[16];
+                    Receive(_clientSocket, hdrBuffer, 0, 16, 300000);
+                    var str = Encoding.Default.GetString(hdrBuffer, 5, 9);
+                    Int32.TryParse(str, out int frameSize);
+
+                    lock (_backLock)
                     {
-                        _backBuffer = new byte[frameSize];
+                        if (_backBuffer.Length != frameSize)
+                        {
+                            _backBuffer = new byte[frameSize];
+                        }
+                        Receive(_clientSocket, _backBuffer, 0, frameSize, 300000);
+                        _frameAvailable.Set();
                     }
-                    Receive(_clientSocket, _backBuffer, 0, frameSize, 300000);
-                    _frameAvailable.Set();
+                }
+                catch(SocketException e)
+                {
+                    log.Error($"{Name}: Socket Exception: {e.Message}");
```

Comment:
Just logging the exceptions isn't enough. The thread should also end at some point (perhaps after 3 consecutive socket exceptions?) or else you'll end up with a tight-loop that's just filling up the log with errors. If the thread does "crash" in this way, the camera object should raise an event or at least expose a flag indicating that the connection has failed. It might be a good idea to leave the object in a failed, but clean, state so that a call to `Connect()` will be able to attempt a reconnect with a new socket and a new worker thread.